### PR TITLE
Exclude files when previously excluded and no wildcards are in play

### PR DESCRIPTION
**Pull request by hvanbakel**
It's not pretty because of the wildcard support we currently have. But if the user is not using a wildcard include then we can generate the appropriate exclusions. Having a wildcard include is probably an exception anyway as it was never really supported in any UI and the current wildcard include is automatic.

### CODE REVIEW

**Comment by hvanbakel** [comment]
I actually thought we were logging this but we must've removed that when wildcards were added.

**Comment by andrew-boyarshin** [comment]
No, it was I who removed that in my IncludeItems refactoring PR, since:
1. It was too verbose (and we didn't have verbosity switch at that time)
2. It required passing additional `ILogger` argument.

**Comment by hvanbakel** [comment]
That's fine, I don't think we need to get that back, it's just that I hadn't noticed it being gone.

File: Project2015To2017/Transforms/FileTransformation.cs [hunk 1]
```diff
@@ -52,13 +52,32 @@ public void Transform(Project definition)
...
+			var wildcardIncludes = keepItems.Where(x => x.Name.LocalName == "Compile").Select(x => x.Attribute("Include")?.Value).Where(x => x != null && x.Contains("*")).ToArray();
+			if (wildcardIncludes.Length > 0)
+			{
+				this.logger.LogWarning("Wildcard include detected, please check for erroneous inclusion of additional files.");
+			}
+			else
+			{
+				var removedIncludes = removeQueue
+					.Where(x => x.Name.LocalName == "Compile")
+					.Select(x => x.Attribute("Update")?.Value)
```
[the remark this answers is not in the document]
Comment (andrew-boyarshin) [reply Project2015To2017/Transforms/FileTransformation.cs]: Strike that. Although `removedIncludes` is not a good name now that I think of it, brings ambiguity to this code block. Is `referencedItems` better?

[the remark this answers is not in the document]
Comment (hvanbakel) [reply Project2015To2017/Transforms/FileTransformation.cs]: I'll apply the rename in my merge

**Comment by hvanbakel** [comment]
I don't know exactly what happened but can someone check what happened in:
beeea00f243b9ee3bd0c6dd38e674f4dfb57a6eb
5e744ff866ca5fde8893bf01eda6aa62db1ff654
e32f8e8ade12dad847ed0157ee524a989822447b

I believe I reverted the commit correctly....

**Comment by andrew-boyarshin** [comment]
@hvanbakel no, bad stuff happened. You reverted what shouldn't be reverted. I'll fix this now.

**Comment by andrew-boyarshin** [comment]
@hvanbakel I have rewritten git history and dropped merge of wrong branches and following wrong revert (in fact, you've additionally reverted 2 PRs). History is now linear without that mistake. To fix git repo locally:
```
> git fetch
> git reset --hard origin/master
```

**Comment by hvanbakel** [comment]
Awesome @andrew-boyarshin this is all still very new to me (fixing branches/wrong merges) so thanks for being the seasoned git person 👍 

So from looking at this again, I don't think I should've done the revert to begin with, it was fine, right?

**Comment by andrew-boyarshin** [comment]
@hvanbakel it was not fine, but was acceptable. Without revert it would preserve all changes though.

**Pull request merged**